### PR TITLE
🧪 Refactor cypress tests (part 3)

### PR DESCRIPTION
**Pull request by Annelein**
Refactors the entire `for-teacher folder` for https://github.com/hedyorg/hedy/issues/4027 :
- `customize-adventure folder` only data-cy and .it() refactoring

**How to test**
I think the best way is to click around the for-teacher page tests and see if they are structured better, readabilty improved and if they cover testing the whole for-teacher page

### CODE REVIEW

**Comment by Felienne** [comment]
ok this one is maybe really failing and not Cypress outage failing: 
image

**Comment by Annelein** [comment]
@Felienne this was meant as a draft!

File: tests/cypress/e2e/for-teacher_page/class_page/other_buttons.cy.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import {loginForTeacher} from '../../tools/login/login.js'
+import { navigateToClass } from "../../tools/classes/class.js";
+import { goToTeachersPage } from '../../tools/navigation/nav.js';
+
+beforeEach(() => {
+  // do we want this test to be run for a second teacher as well?
```
Comment (Annelein) [inline tests/cypress/e2e/for-teacher_page/class_page/other_buttons.cy.js:6]: Do we want the whole other_buttons test, to also be tested for a second teacher?

File: tests/cypress/e2e/for-teacher_page/customize_class_page/preview_class.cy.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { createClass, openClassView, selectLevel } from '../../tools/classes/class.js';
+import { loginForTeacher } from '../../tools/login/login.js'
+import { goToTeachersPage } from '../../tools/navigation/nav.js';
+
+// Do we also want this to be tested for a second_teacher?
```
Comment (Annelein) [inline tests/cypress/e2e/for-teacher_page/customize_class_page/preview_class.cy.js:5]: Do we also want this to be tested for a second_teacher?

File: tests/cypress/e2e/for-teacher_page/customize_class_page/quiz_parsons.cy.js [hunk 3]
```diff
@@ -1,151 +1,97 @@
 import { loginForStudent, loginForTeacher } from '../../tools/login/login.js'
-import { ensureClass } from "../../tools/classes/class";
+import { ensureClass, openClassView, removeCustomizations, selectLevel } from "../../tools/classes/class";
+import { goToHedyLevel2Page, goToHedyLevel5Page } from '../../tools/navigation/nav.js';
 
+// Do we also want this to be tested for a second_teacher?
```
Comment (Annelein) [inline tests/cypress/e2e/for-teacher_page/customize_class_page/quiz_parsons.cy.js:5]: Do we also want this to be tested for a second_teacher?

File: tests/cypress/e2e/for-teacher_page/class_page/other_buttons.cy.js (hunk 1 above)
Comment (jpelay) [reply tests/cypress/e2e/for-teacher_page/class_page/other_buttons.cy.js:6]: I think so. It'll make the test slower (and I think you'd need to make a fresh class before each set of test: one for the teacher and one for the second teacher), but tests are so useful already that I think the added time is worth it! The same answer goes for every test where this question was raised.

**Comment by Annelein** [comment]
@jpelay added the tests for second teachers! so its ready

**Review by jpelay — APPROVED** [review]
Everything seems to be working fine! Great job by grouping some of these tests and simplifying them

**Pull request merged**